parsers/promise-with-wait.ts

```diff
@@ -0,0 +1,20 @@
+export async function promiseWithTimeout<T>(timeoutMs: number, promise: () => Promise<T>): Promise<T> {
```

Comment:
Handy thing that will handle timeouts when promise is executed... it is used to time out the parsing operation when it takes too long.